fedimint-server/src/consensus/aleph_bft/data_provider.rs

```diff
@@ -100,6 +108,20 @@ impl aleph_bft::DataProvider<UnitData> for DataProvider {
 
         assert!(bytes.len() <= ALEPH_BFT_UNIT_BYTE_LIMIT);
 
+        self.timestamp_sender
+            .send((Instant::now(), get_citem_bytes_chsum(&bytes)))
+            .await
+            .ok();
+
         Some(UnitData::Batch(bytes))
     }
 }
+
+/// Calculate a cheap chesum of an encoded citem, as a
+/// XOR of its len and first 8 bytes.
```

Comment:
Why only the first 8 bytes :sob: hashing isn't that expensive, even if it was we could use a CRC instead of SHA256.

Reply:
Ehhh, the whole thing will be exercised once a year, max, because what are the chances that long running daemon like `fedimintd` gets restarted right after it broadcasted an item, and then after the restart but before receiving such item, it also broadcasted a different item already? And a accidental conflict here just means prom metrics will report a single or a handful of prom metric with some slightly off time. It really just doesn't matter at all, so why do we worry so much?

And the code here is exactly on the codepath we're trying to measure and keep well oiled. Pulling CRC deps or doing some extra hashing is IMO optimizing for the irrelevant.

Plus if I'm prove wrong, this can be changed extra easily.

Reply:
Changed. Most important is to optimize for landing stuff.